### PR TITLE
Make several improvements to the way the CI runs the tests.

### PR DESCRIPTION
**Pull request by cbezault**
1. Discover and run all the libcxx, std, and tr1 as a single step.
This minimizes the amount of single threaded time we take to run the
tests.
2. Make the tests have a timeout in the CI. (In order to do this I need
to `pip install psutil`)
3. Shuffle the order the tests are run in the CI.
4. Don't require vcpkg.exe to run the build
5. Move the build output directory of the support tools to be on the
D: drive.
6. A JSON results file will now get created when running the `ctest` command and no other LIT flags are specified.

### CODE REVIEW

**Comment by AdamBucior** [comment]
What's the reason for shuffling the order of the tests? The log looks really messy because of this.

**Comment by BillyONeal** [comment]
> What's the reason for shuffling the order of the tests? The log looks really messy because of this.

It ensures tests don't pass because they have accidental dependence on the order in which they run.

**Comment by cbezault** [comment]
The log looks messy but you can still view all the tests in the tests tab in a nice format.

**Comment by CaseyCarter** [comment]
Apparently toggling draft status on doesn't prevent spam in the inboxes of folks subscribed to the PR; I have an email notification for each of the last five commits.

**Comment by cbezault** [comment]
Ew, that's unfortunate

**Comment by cbezault** [comment]
Added benefit to shuffling is increasing the likelihood the multi-threaded tr1 tests don't stomp all over each other without actually bothering to fix the underlying issue. We could see a ~1min test time decrease just from that.

**Comment by StephanTLavavej** [comment]
https://github.blog/2019-02-14-introducing-draft-pull-requests/ says "Also, if you have a CODEOWNERS file in your repository, a draft pull request will suppress notifications to those reviewers until it is marked as ready for review." but perhaps that only affects code owners who aren't watching literally everything in a repository.

File: README.md [hunk 1]
```diff
@@ -248,11 +248,11 @@ under a category in libcxx, or running a single test in `std` and `tr1`.
 
 C:\STL\build>ctest -V
 
-:: This command will run all of the std testsuite.
+:: This command will also run all of the testsuites
```
Comment (StephanTLavavej) [inline README.md:251]:
```suggestion
:: This command will also run all of the testsuites.
```

File: README.md [hunk 2]
```diff
@@ -271,19 +271,21 @@ C:\STL\build>python tests\utils\stl-lit\stl-lit.py ..\llvm-project\libcxx\test\s
 
 ### CTest
 
-When running the tests via CTest each of the testsuites is considered to be a single test. If any single test in a
-testsuite fails, CTest will report the test which represents that testsuite as failed.
+When running the tests via CTest all of the testsuites are considered to be a single test. If any single test in a
```
Comment (StephanTLavavej) [inline README.md:274]:
```suggestion
When running the tests via CTest, all of the testsuites are considered to be a single test. If any single test in a
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -28,12 +28,14 @@ stages:
...
+                rmdir /S /Q "D:\tools"
+              )
               call "%PROGRAMFILES(X86)%\Microsoft Visual Studio\2019\Preview\Common7\Tools\VsDevCmd.bat" ^
               -host_arch=amd64 -arch=amd64 -no_logo
               cmake -G Ninja -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=Release ^
-              -S $(Build.SourcesDirectory)\tools -B $(Build.ArtifactStagingDirectory)\tools
-              cd $(Build.ArtifactStagingDirectory)\tools
-              cmake --build .
+              -S $(Build.SourcesDirectory)\tools -B D:\tools
+              cmake --build D:\tools
```
Comment (StephanTLavavej) [inline azure-pipelines.yml:38]: Should `D:\tools` be a named variable at the top?

File: azure-devops/run-build.yml [hunk 4]
```diff
@@ -21,7 +26,7 @@ jobs:
       clean: true
       submodules: true
     - task: Cache@2
-      displayName: Cache vcpkg
+      displayName: Restore Cached vcpkg Installed Directory
```
[the remark this answers is not in the document]
Comment (cbezault) [reply azure-devops/run-build.yml:29]: Yeah that was an awkward display name. The directory is really called `installed`. I could just change it back to cache `vcpkg`?

File: README.md [hunk 5]
```diff
@@ -248,11 +248,11 @@ under a category in libcxx, or running a single test in `std` and `tr1`.
 
 C:\STL\build>ctest -V
 
-:: This command will run all of the std testsuite.
+:: This command will also run all of the testsuites.
 
-C:\STL\build>ctest -R std
+C:\STL\build>python tests\utils\stl-lit\stl-lit.py ..\llvm-project\libcxx\test ..\tests\std ..\tests\tr1
```
Comment (StephanTLavavej) [inline README.md:253]: There will be a merge conflict with #783 here.

**Comment by StephanTLavavej** [comment]
Thanks for these improvements, Curtis! I'm merging this first and will resolve the conflict in #783.

**Pull request merged**